Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/dialogs/GamePropertiesDialog.java

```diff
@@ -118,4 +122,32 @@ private void clearGameSettings(String gameId)
       Toast.makeText(getContext(), "No game settings to delete", Toast.LENGTH_SHORT).show();
     }
   }
+
+  private boolean recursivelyDeleteGameProfiles(@NonNull final File file, String gameId)
+  {
+    boolean hadGameProfiles = false;
+    File[] files = file.listFiles();
```

Comment:
`listFiles` should be called after checking `isDirectory`. The `isDirectory` check is a bit pointless otherwise.